packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -454,6 +460,93 @@ class RenderEditable extends RenderBox {
     );
   }
 
+  void _handleMoveCursorForwardByWord(bool extentSelection) {
+    final TextRange currentWord = _textPainter.getWordBoundary(_selection.extent);
+    if (currentWord == null)
+      return;
+    final TextRange nextWord = _getNextWord(currentWord.end);
+    if (nextWord == null)
+      return;
+    final int baseOffset = !extentSelection ? nextWord.start : _selection.baseOffset;
```

Comment:
This might be easier to comprehend if you switch it around and remove the negation?

Reply:
Done